kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/AddFile.java

```diff
@@ -125,7 +129,8 @@ public static Row createAddFileRow(
     baseRowId.ifPresent(id -> fieldMap.put(FULL_SCHEMA.indexOf("baseRowId"), id));
     defaultRowCommitVersion.ifPresent(
         version -> fieldMap.put(FULL_SCHEMA.indexOf("defaultRowCommitVersion"), version));
-    stats.ifPresent(stat -> fieldMap.put(FULL_SCHEMA.indexOf("stats"), stat.serializeAsJson()));
+    stats.ifPresent(
+        stat -> fieldMap.put(FULL_SCHEMA.indexOf("stats"), stat.serializeAsJson(tableSchema)));
```

Comment:
serializeAsJson takes the `dataSchema` but here we are passing in the `tableSchema` -- just calling this out. I think this is fine?